streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -1458,6 +1458,14 @@ protected boolean maybeUpdateSubscriptionVersion(final int receivedAssignmentMet
 
     @Override
     public void onAssignment(final Assignment assignment, final ConsumerGroupMetadata metadata) {
+        final Set<Task> tasksWithOpenTransactions = taskManager.allOwnedTasks()
```

Comment:
Seem we would need to check if EOS is enabled or not? Or would we want to commit also for the ALOS case?

Reply:
Ah. Good point, no need to commit in ALOS.